### ISSUE TITLE
Can't deploy easily with Android Studio 2.1

### ISSUE THREAD

**Issue by jimbeveridge** [flutter/flutter#3924]
This bug is for `hello_android`-style projects where Flutter is a view in a Java app on Android.

Android Studio 2.1 added "Hot Run", which does not recognize that the resources changed for Dart and so doesn't redeploy properly. A complete rebuild is always required, which takes 5 to 20 seconds. This slows down the development cycle drastically.

Android Studio 2.1 is now the current "stable" release.

I've already updated the gradle files for 2.1 in  6e488249916d2bed08c278d594e69487fee7358d.

**Comment by sethladd** [comment]
cc @devoncarew

**Comment by jimbeveridge** [comment]
[From Jason Simmons]

I was able to disable Instant Run in Android Studio by going to File
-> Settings -> Build/Execution/Deployment -> Instant Run and unchecking the first checkbox.

With this, the IDE workflow should deploy the right code as needed. Clicking the Run button will run a Gradle build, which will rebuild Java and/or Dart code (if either has been modified), repackage the
APK, and install it on the device.

FWIW, I set up my project by cloning hello_android and importing that into Android Studio.  The outputs of the build are showing up in the same place that "gradlew build" would use when run from the command line. (I haven't tried adding Flutter to a project generated by Android Studio).
